### PR TITLE
nixos/doc/rl-2411: improve wording of macOS version warning

### PR DESCRIPTION
**Pull request by emilazy**
## Description of changes

Fixes: f1c3597d95027fdd8f786716e02c0ed4167842d9

cc @SigmaSquadron @RossComputerGuy

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SigmaSquadron — APPROVED** [review]
LGTM. Thanks!

File: nixos/doc/manual/release-notes/rl-2411.section.md [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
 ## Highlights {#sec-release-24.11-highlights}
 
-- **This will be the latest version of Nixpkgs to support macOS 10.12 (Sierra).**
+- **This will be the last release of Nixpkgs to support macOS 10.12 (Sierra) to 10.15 (Catalina).**
   Starting with release 25.05, the minimum supported version will be macOS 11 (Big Sur).
-  From that point on, packages may or may not work on older releases of macOS.
-  Users on old macOS versions should consider upgrading to a supported OS release (potentially using [OpenCore Legacy Patcher](https://dortania.github.io/OpenCore-Legacy-Patcher/) for old hardware) or installing NixOS.
+  From that point on, packages may or may not work on older versions of macOS.
```
Comment (AndersonTorres) [inline nixos/doc/manual/release-notes/rl-2411.section.md:9]:
```suggestion
  From that point on, we will provide no guarantees that packages will work on older versions of macOS.
```

**Comment by AndersonTorres** [comment]
Not my OS, but the support you provided for such an old platform was impressive!

File: nixos/doc/manual/release-notes/rl-2411.section.md (hunk 1 above)
Comment (emilazy) [reply nixos/doc/manual/release-notes/rl-2411.section.md]: Pushed a slightly different tweak.

**Comment by emilazy** [comment]
It was mostly accidental, honestly. In the past the SDK version and deployment target were bumped more regularly, but it stalled out due to maintainer resources until the release of Apple Silicon machines, and the status quo on `x86_64-darwin` stuck. We’re not sure if anyone was actually using these old versions, but I’m happy if we were able to help people stuck on old Mac hardware, and hopefully we can continue to do so through the OpenCore Legacy Patcher or NixOS. Going forward we’re likely to drop support for versions on a more regular cadence again, synchronized between `aarch64-darwin` and `x86_64-darwin`.

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**